client/api.c

```diff
@@ -980,6 +981,352 @@ TDNFRepoList(
     goto cleanup;
 }
 
+static int
```

Comment:
Return type is uint32_t

Reply:
No, it needs to be `int`. This callback is used for `ntfw()`, and needs to match the prototype. See `man 3 nftw`:
```
       int nftw(const char *dirpath,
               int (*fn) (const char *fpath, const struct stat *sb,
                          int typeflag, struct FTW *ftwbuf),
               int nopenfd, int flags);
```

Reply:
In that case, should we change the dwError to int or handle the error differently? 
As per man of ntfw():
`If fn() returns nonzero, then the tree walk is terminated and the value returned by fn() is returned as the result of ftw() or nftw().`
In case of error from this function, the return value will not be < 0 which is used in the code here:
` if (nftw(pszRepoDir, _rm_rpms, 10, FTW_DEPTH|FTW_PHYS) < 0)`

Reply:
Yes, you are right that we should bail line 1260 on any non-zero return. A -1 return is presumably a system error in which case we need to check `errno` , but a value greater zero is a TDNF error generated by the callback. Good point!

Reply:
Fixed.